scipy/_lib/_testutils.py

```diff
@@ -215,3 +216,25 @@ def _get_mem_available():
             return info['memfree'] + info['cached']
 
     return None
+
+
+def pytest_warns():
+    """
+    In pytest 6.x, one can use `pytest.warns(None)`, but `pytest.warns()`
+    will raise a TypeError.  In pytest 7.x, `pytest.warns(None)` triggers
+    a DeprecationWarning, with a message that recommends using
+    `pytest.warns()`.
+
+    This function is a compatibility shim that allows us to replace
+    `pytest.warns(None)` with `pytest_warns()`.  This function will
+    work with pytest 6 and 7.
+
+    Once version 7 of pytest is the minimum support version, we can
+    remove this function and replace its use with `pytest.warns()`.
+    """
+    import pytest
+
+    if int(pytest.__version__.split('.')[0]) > 6:
```

Comment:
I realize this is a shim, would it be more readable to use:
```
from scipy._lib._pep440 import Version
...
if Version(pytest.__version__) > Version('6.0'):
``` 
?

Reply:
Also, did you mean to close the pull request? 

I'm just seeing that it was closed after marking as draft a couple hours ago.

Reply:
@rlucas7, thanks for taking a look.  I closed the PR because the change wasn't *quite* as trivial as I thought, and I had other eggs to fry.  I created an issue about the change required for pytest 7.0.0: https://github.com/scipy/scipy/issues/15186